hyperapp.d.ts

```diff
@@ -94,12 +116,15 @@ export interface View<State, Actions> {
  * @returns The actions wired to the application.
  * @memberOf [App]
  */
-export function app<State, Actions>(
+export function app<
+  State extends StateType,
+  AppActions extends ActionsType<State, AppActions>
+>(
   state: State,
-  actions: ActionsType<State, Actions>,
-  view: View<State, Actions>,
+  actions: ActionsType<State, AppActions>,
```

Comment:
Also `AppActions` here.

Reply:
Same reason

Reply:
I don't understand.

Reply:
You need to understand how Generics work. A generic can be named with a single letter, which is usually done, but also as a name. In this case we could even prefix generics with a G so it isn't ambiguous. The Typescript compiler needs to know the difference between a `Type` and a `Generic`. Read up more here: https://www.typescriptlang.org/docs/handbook/generics.html

Reply:
I will try. BTW you gave me a good example with StateType, we made progress there, but there was none for actions, hence we are still stuck.

Reply:
Have a look here, we want the generic type for `Actions` in `app()` (ie `app<MyStateType, MyActionsType>`) to be constraint to having only functions of the following signature:
- `(data?) => state | ((state?, actions?) => state)`. Have a look here at 'Generic Constraints' https://www.typescriptlang.org/docs/handbook/generics.html#generic-constraints